src/TodosContext.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from 'react';
+
+export type Todo = {
+  title: string,
+  id: number,
+  completed: boolean,
+  editing: boolean,
+};
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const todos: Todo[] = [];
+
+function prepareTodos(todosList: Todo[], status: string): Todo[] {
+  return todosList.filter((todo) => {
+    switch (status) {
+      case 'completed':
+        return todo.completed;
+      case 'active':
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+}
```

Comment:
No need to filter in case status is all - it may cause performance issue
```suggestion
function prepareTodos(todosList: Todo[], status: string): Todo[] {
  switch (status) {
      case 'completed':
        return todosList.filter((todo) =>todo.completed);
      case 'active':
        return todosList.filter((todo) =>!todo.completed);
      default:
        return todosList;
    }
}
```